File: backend/models/Settlement.js
const mongoose = require("mongoose")

const settlementSchema = new mongoose.Schema({
  expenseId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Expense",
    required: true
  },
  personName: {
    type: String,
    required: true
  },
  amount: {
    type: Number,
    required: true
  },
  settledAt: {
    type: Date,
    default: Date.now
  },
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "User"
  }
})

module.exports = mongoose.model("Settlement", settlementSchema)

File: backend-springboot/src/main/java/com/splitwiseplus/model/Settlement.java
package com.splitwiseplus.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
import com.splitwiseplus.util.ObjectIdDeserializer;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.util.Date;

@Document(collection = "settlements")
@Data
public class Settlement {
    @Id
    @JsonProperty("_id")
    private String id;
    
    @JsonSerialize(using = ToStringSerializer.class)
    @JsonDeserialize(using = ObjectIdDeserializer.class)
    private org.bson.types.ObjectId expenseId;
    
    private String personName;
    private String personEmail;
    private Double amount;
    private Date settledAt = new Date();
    
    @JsonSerialize(using = ToStringSerializer.class)
    @JsonDeserialize(using = ObjectIdDeserializer.class)
    private org.bson.types.ObjectId userId;
}
